diff --git a/src/frameworks/foundation/ns_data.rs b/src/frameworks/foundation/ns_data.rs
index cc40c2e5..63156361 100644
--- a/src/frameworks/foundation/ns_data.rs
+++ b/src/frameworks/foundation/ns_data.rs
@@ -184,6 +184,21 @@ pub const CLASSES: ClassExports = objc_classes! {
     env.fs.write(GuestPath::new(&file), slice).is_ok()
 }
 
+- (bool)writeToFile:(id)path // NSString*
+            options:(NSUInteger)_options // NSDataWritingOptions
+              error:(MutPtr<id>)error { // NSError**
+    // The options (atomic write, file protection) don't matter here.
+    let ok: bool = msg![env; this writeToFile:path atomically:true];
+    if !ok {
+        log!("Warning: [(NSData*){:?} writeToFile:options:error:] failed", this);
+        // TODO: return a real NSError
+        if !error.is_null() {
+            env.mem.write(error, nil);
+        }
+    }
+    ok
+}
+
 - (())dealloc {
     let &NSDataHostObject { bytes, free_when_done, .. } = env.objc.borrow(this);
     if !bytes.is_null() && free_when_done {
